staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -279,6 +279,12 @@ func (w *watchCache) processEvent(event watch.Event, resourceVersion uint64, upd
 		wcEvent.PrevObjFields = previousElem.Fields
 	}
 
+	if wcEvent.Type != watch.Bookmark {
+		internalEvent := *wcEvent
+		setCachingObjects(&internalEvent, w.versioner)
```

Comment:
Is this changing the watch cache so that all the events held in it retain their serialized forms in all versions and formats? If so, doesn't adding caching at this level dramatically increase the retained memory?

When serialization caching was added originally, the only reason it seemed reasonable was because the retention lifetime was limited to the sending of a single event to (possibly) multiple watchers. I am very unsure whether the memory footprint this change causes will be acceptable at large scale.

Reply:
+1 to Jordan, I agree this change will result in reduced cpu usage in some cases, but that comes with (potentially very non-negligible cost) for *all* cases.

For the particular usecase you mentioned, CiliumEndpointSlice was the response to optimize the stack:
https://docs.cilium.io/en/latest/network/kubernetes/ciliumendpointslice/

I would be hesitant to accepting this change, until we have a clear and widely accepted upper-bound for memory increase, which isn't clear at this point.